presto-geospatial/src/test/java/com/facebook/presto/plugin/geospatial/TestGeoFunctions.java

```diff
@@ -126,6 +126,23 @@ public void testSTCentroid()
         assertFunction("ST_AsText(ST_Centroid(ST_GeometryFromText('POLYGON ((0 0, 0 5, 5 5, 5 0, 0 0), (1 1, 1 2, 2 2, 2 1, 1 1))')))", VARCHAR, "POINT (2.5416666666666665 2.5416666666666665)");
     }
 
+    @Test
+    public void testSTConvexNull()
+    {
+        // test Invalid types
+        assertInvalidFunction("ST_AsText(ST_ConvexNull(ST_GeometryFromText('LINESTRING EMPTY')))", VARCHAR, "ST_Polygon only applies to MULTI_LINE_STRING or MULTI_POINT or MULTI_POLYGON or GEOMETRY_COLLECTION\n. Input type is: LINE_STRING");
+        assertInvalidFunction("ST_AsText(ST_ConvexNull(ST_GeometryFromText('POINT (3 5)')))", VARCHAR, "ST_Polygon only applies to MULTI_LINE_STRING or MULTI_POINT or MULTI_POLYGON or GEOMETRY_COLLECTION\n. Input type is: POINT");
+        assertInvalidFunction("ST_AsText(ST_ConvexNull(ST_GeometryFromText('LINESTRING (1 1, 2 2, 3 3)')))", VARCHAR, "ST_Polygon only applies to MULTI_LINE_STRING or MULTI_POINT or MULTI_POLYGON or GEOMETRY_COLLECTION\n. Input type is: LINESTRING");
+        assertInvalidFunction("ST_AsText(ST_ConvexNull(ST_GeometryFromText('POLYGON ((1 1, 1 4, 4 4, 4 1)')))", VARCHAR, "ST_Polygon only applies to MULTI_LINE_STRING or MULTI_POINT or MULTI_POLYGON or GEOMETRY_COLLECTION\n. Input type is: POLYGON");
+
+        // test valid types
+        assertFunction("ST_AsText(ST_ConvexNull(ST_GeometryFromText('MULTILINESTRING EMPTY')))", VARCHAR, "POLYGON EMPTY");
+        assertFunction("ST_AsText(ST_ConvexNull(ST_GeometryFromText('MULTILINESTRING (1 1, 5 1), (2 4, 4 0) (2 -4, 4 4) (3 -2, 4 -3)')))", VARCHAR, "POLYGON (1 1, 2 4, 4, 4, 5 1, 4 -3, 2 -4)");
+        assertFunction("ST_AsText(ST_ConvexNull(ST_GeometryFromText('MULTIPOINT (0 2, 1 0, 3 0, 4 0, 4 2, 2 2, 2 4)')))", VARCHAR, "POLYGON (0 2, 2 4, 4 2, 4 0, 1 0)");
+        assertFunction("ST_AsText(ST_ConvexNull(ST_GeometryFromText('MULTIPOLYGON (0 3, 2 0, 3 6), (2 1, 2 3, 5 3, 5 1), (1 7, 2 4, 4 2, 5 6, 3 8)')))", VARCHAR, "POLYGON (0 3, 1 7, 3 8, 5 6, 5 1, 2 0)");
+        assertFunction("ST_AsText(ST_ConvexNull(ST_GeometryFromText('GEOMETRYCOLLECTION (POINT (2 3), LINESTRING (2 8, 7 10)), POINT (8 10), POLYGON (4 4, 4 8, 9 8, 6 6, 6 4, 8 3, 6 1), POINT (4 2), LINESTRING (3 6, 5 5), POLYGON (7 5, 7 6, 8 6, 8 5)')))", VARCHAR, "POLYGON (2 3, 2 8, 7 10, 8 10, 9 8, 8 3, 6 1)");
```

Comment:
Could you add tests for single geometries, both convex and non-convex?

Reply:
@mbasmanova Hi, I think I understand the concept of convex hull for polygon. But what's the convex hull for a point and a line string? I don't know if it's applicable.